Script Overhaul in relation to issue # 1

Leatherman-Security opened the pull request:
Best of luck!

https://github.com/simeononsecurity/.NET-STIG-Script/issues/1

Leatherman-Security committed:
Entire Script Overhaul

Entire Script Overhaul

Leatherman-Security committed:
Rename of machine.config

Rename of machine.config

simeononsecurity committed:
Update secure.machine.config

simeononsecurity committed:
Correct $DotNetVersion References

simeononsecurity committed:
Fix Formatting

simeononsecurity committed:
Save Machine.Conf to Origin Directory

simeononsecurity committed:
Revert

simeononsecurity commented:
Fixed a few things, but it seems that the final machine.conf isn't saved in the originating directory. It's saved in the script root. Can you look into it @Leatherman-Security ?

Leatherman-Security committed:
Corrected Set-SecureConfig param and did Out-Nulls

Corrected Set-SecureConfig param, it was adding part of the path back in after it was already being passed the full path

Added | Out-Nulls to registry changes to reduce noise back to operator.

Leatherman-Security committed:
Machine.config missing in Param pass

Machine.config missing in Param pass

Leatherman-Security committed:
Improper Function call

With .fullname being added to the param object, it needed to be given a chance to execute and then be placed inside the string as before object was just a string and with change it has to be invoked first before being ran.

Added in a few more | Out-Nulls to reduce channel noise

simeononsecurity commented:
Glad to see those changes. Now we just need to make sure the lines are implemented in the correct sections rather than just appending the machine.conf file. If we manage that. This will be G2G.

Leatherman-Security committed:
Added in basic element checks

Now checks for first XML element presence before adding it in.
If it is there, it checks one element deeper, if it is not present it adds it.

Need to have it check the content of that element to see if it is desired content as well.  Will need to figure out how to build out a function that can be continuously called in a loop, appending unique elements each time to check all content.

Spacing might be slightly off...

simeononsecurity committed:
Update README.md

simeononsecurity committed:
Update secure.machine.config

simeononsecurity commented:
@Leatherman-Security Any updates?

Leatherman-Security committed:
Update sos-.net-4-stig.ps1

Corrected a few incorrect pathing calls for start-process

Leatherman-Security committed:
Update sos-.net-4-stig.ps1

Larger scale edits.
 Fixed incorrect appending option
 Added individual element overwrite if value is there.  A long term do out would be to test the contents of it and only overwrite if it doesn't matches.

simeononsecurity committed:
Update sos-.net-4-stig.ps1

simeononsecurity committed:
Formatting and Description

Leatherman-Security committed:
Update sos-.net-4-stig.ps1

Two big hold ups turned out to be COM variable sometimes not resetting despite being set to a new object.  
    Fixed by declaring said variables to be blank initially.
    Additionally set powershell to run the garbage collector before starting a new instance of working with com objects.

Leatherman-Security committed:
Update sos-.net-4-stig.ps1

Realized I had removed a few Clarification items added in a previous version. Added them back in

Leatherman-Security committed:
Update sos-.net-4-stig.ps1

Line correction

simeononsecurity committed:
AvoidUsingCmdletAliases

https://github.com/PowerShell/PSScriptAnalyzer/blob/master/RuleDocumentation/AvoidUsingCmdletAliases.md

simeononsecurity commented:
Seems there is a issue were some of the configs not need be applied to lower versions of .net

- https://docs.microsoft.com/en-us/dotnet/framework/configure-apps/file-schema/runtime/enforcefipspolicy-element 
  - (2.0 or higher)
- https://docs.microsoft.com/en-us/dotnet/framework/configure-apps/file-schema/runtime/loadfromremotesources-element 
  - (4.0 or higher)
- https://docs.microsoft.com/en-us/dotnet/framework/configure-apps/file-schema/runtime/netfx40-legacysecuritypolicy-element
  - (4.0 or higher)
- https://docs.microsoft.com/en-us/dotnet/framework/configure-apps/file-schema/runtime/etwenable-element 
  - (Doesn't specify. Assuming 3.0 or higher because it mentions Vista)
- https://docs.microsoft.com/en-us/dotnet/framework/configure-apps/file-schema/network/defaultproxy-element-network-settings 
  - (Doesn't specify.)

simeononsecurity committed:
Split Conf Files

https://github.com/simeononsecurity/.NET-STIG-Script/pull/2#issuecomment-865513194

simeononsecurity committed:
Version Specific Config Files

https://github.com/simeononsecurity/.NET-STIG-Script/pull/2#issuecomment-865513194

simeononsecurity committed:
Fixed syntax

simeononsecurity committed:
Remove Error Handling and Cleaned Output

simeononsecurity committed:
Improve Output Verbosity

Leatherman-Security committed:
Update sos-.net-4-stig.ps1

Cleaned up a few lines.
Added If statement to catch when there is not a machine.config file in .net framework
Added line to ensure machine.config file is CLOSED before saving changes to it.
Moved if statement for catching empty fields up a step to prevent error creation when attempting to import said non-existent field.

Leatherman-Security committed:
Update sos-.net-4-stig.ps1

Removed misplaced comment.
Added in comments for clarification.
Adjusted Contributor tag/information
Cleaned up a few uneeded troubleshooting functions.
Updated wording on line 85

simeononsecurity commented:
Bugs fixed and comments added. LG2M.
Thx for the hard work!
